File: QE-selenium/java/Activity13.java
package demos;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;

import java.util.List;

public class Activity13 {

    public static void main(String[] args) {

        WebDriver driver = new FirefoxDriver();

        try {
            driver.get("https://training-support.net/webelements/tables");

          
            List<WebElement> rows = driver.findElements(By.xpath("//table/tbody/tr"));
         
            List<WebElement> cols = driver.findElements(By.xpath("//table/thead/tr/th"));

            System.out.println("rows: " + rows.size());
            System.out.println("cols: " + cols.size());

            int i = 1;
            for (WebElement row : rows) {
                if (i == 3) {
                    System.out.println(row.getText());
                }
                i++;
            }

            String secondRowSecondColumn =
                    driver.findElement(By.xpath("//table/tbody/tr[2]/td[2]")).getText();

            System.out.println(secondRowSecondColumn);

            System.out.println(driver.getTitle());

        } finally {
            driver.quit();
        }
    }
}

File: QE-selenium/python/Activity13.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.common.action_chains import ActionChains
from selenium.webdriver.common.keys import Keys
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC

with webdriver.Firefox() as driver:
    driver.get("https://training-support.net/webelements/tables")
    rows = driver.find_elements(By.XPATH,"//table/tbody/tr")
    cols = driver.find_elements(By.XPATH,"//table/thead/tr/th")
    print("rows: " , len(rows))
    print("cols: " , len(cols))
    i=1
    for row in rows:
        if i == 3:
            print(row.text)
        i+=1
    
    second_row_second_column = driver.find_element(By.XPATH,"//table/tbody/tr[2]/td[2]").text
    print(second_row_second_column)
    print(driver.title)
    driver.quit()
